src/zimCheck.h

```diff
@@ -85,8 +85,9 @@ void getDependencies(std::string page, std::vector <std::string> *links)
                 {
                     i++;
                 }
-                length=i-startingPoint;
-                links->push_back(page.substr(startingPoint,length));
+                std::string url = page.substr(startingPoint, i-startingPoint);
+                if (std::regex_match(url, std::regex("http(s|)://")))
```

Comment:
Are you sure about this ?

`std::match` only considers full matches.
See how the following code behaves :
```cpp
#include <string>
#include <regex>
#include <iostream>
int main() {
  std::string url = "http://foo/bar.html";
  std::cout << "match " << std::regex_match(url, std::regex("http(s|)://")) << std::endl;
  std::cout << "search " << std::regex_search(url, std::regex("http(s|)://")) << std::endl;
 }
```
>>>
match 0
search 1

Reply:
I have renamed the PR and fixed the regex (beside other things). I have discovered an other bug which needs to be fixed too https://github.com/openzim/zim-tools/issues/32... but this is a bit more complex.